### ISSUE TITLE
lc.bin().to_pandas() failed with TypeError: data type '>I4' not understood

### ISSUE THREAD

**Issue by orionlee** [lightkurve/lightkurve#1043]
#### Problem description

Calling `lc.bin().to_pandas()` failed with `TypeError: data type '>I4' not understood` ;  at least for the TESS 2-minute cadence files I've tried.

- As a workaround, if I remove the integer columns from the binned lc (`cadenceno` and `quality`), it works.
- I tried to create a standalone case and have not yet been able to reproduce it.

#### Example

```python
import lightkurve
# insert code here ...

# lc is a typical TESS 2-minute cadence product, e.g., TIC197787155
> lc_b = lc.bin(time_bin_size=20 * u.min)
> lc_b.to_pandas()

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-458-8be46a828db6> in <module>
----> 1 lc_b.to_pandas()

C:\dev\lightkurve\src\lightkurve\lightcurve.py in to_pandas(self, **kwargs)
   2049         -------
   2050         dataframe : `pandas.DataFrame`
-> 2051             A data frame indexed by `time`.
   2052         """
   2053         df = super().to_pandas(**kwargs)
...
c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\pandas\core\dtypes\common.py in pandas_dtype(dtype)
   1761     # raise a consistent TypeError if failed
   1762     try:
-> 1763         npdtype = np.dtype(dtype)
   1764     except SyntaxError as err:
   1765         # np.dtype uses `eval` which can raise SyntaxError

TypeError: data type '>I4' not understood
```
Reproducing it in a standalone lightcurve: so far I have not been able reproduce the error.
I have included similar integer `cadenceno` and `quality` columns in the lightcurve. No error is emitted.

```python
lc = lk.LightCurve(time=100 + np.arange(12), flux=999.1 + np.arange(12), flux_err=2 ** 0.5 * np.ones(12))
lc['cadenceno'] = np.arange(12)
lc['quality'] = 12345 * np.ones(12)
lc_b = lc.bin(time_bin_size=3 * u.day)
lc_b.to_pandas()
```

#### Environment

-  platform (e.g. Linux, OSX, Windows): Windows
-  lightkurve version (e.g. 1.0b6): 2.0.9
-  pandas version: 1.1.5
-  astropy version: 4.1
-  installation method (e.g. pip, conda, source): pip

**Comment by orionlee** [comment]
Update: I tried some Kepler products - it had similar problems, e.g., 

```python
lc = lk.search_lightcurve('KIC 3733346', author='Kepler', exptime=60).download()
lc_b = lc.bin(time_bin_size=30*u.min)
lc_b.to_pandas()
```

Again, the issue can be avoided if integer columns are removed (Kepler product has an additional `sap_quality` in addition to the above two).

**Comment by orionlee** [comment]
Narrowed down the problem. I don't know if it should be fixed in lightkurve or upstream astropy / pandas.

At is root, the error occurs when:
1. the `str(dtype)` of the column is '>i4', and
2. the column is a `MaskedColumn`

Using `cadenceno` column from typical TESS / Kepler objects as an example,
1. the `dtype` is 32 bit integer, with `str(dtype)` is '>i4', and
2. after calling `lc.bin()`, the `cadenceno` column of the binned lc is a `MaskedColumn`
3. when calling `to_pandas()` on the binned lc, the above error occured.

Possible fixes:
1. In `lightkurve`, change `cadenceno`'s dtype to `np.int32`, its string representation is `int32`, possibly upon initial reading the FIS file.
2. In `astropy` `Table.to_pandas()`, convert '>i4' to `Int32` before passing it to pandas. https://github.com/astropy/astropy/blob/v4.1/astropy/table/table.py#L3430
3. Make `pandas` understand `>I4`

Thoughts? I don't quite know where the fix logically belongs - e.g., I am not familiar with the possible difference between plain `np.int32` and `>i4` integer type.

**Comment by orionlee** [comment]
The fix in astropy is to be included in astropy 4.3.

**Comment by orionlee** [comment]
Fixed with lightkurve 2.1, as it requires astropy 5.x, which includes the fix.
